api/service/v1alpha1/service.proto

```diff
@@ -210,10 +216,12 @@ message ListStagesResponse {
 message GetStageRequest {
   string project = 1;
   string name = 2;
+  RawFormat format = 3;
 }
 
 message GetStageResponse {
-  github.com.akuity.kargo.api.v1alpha1.Stage stage = 1;
+  optional github.com.akuity.kargo.api.v1alpha1.Stage stage = 1;
+  optional bytes raw = 2;
```

Comment:
I think we can make this more safer by leveraging [oneOf](https://protobuf.dev/programming-guides/proto3/#oneof) here 🤔 

```suggestion
  oneof data {
    optional github.com.akuity.kargo.api.v1alpha1.Stage stage = 1;
    optional bytes raw = 2;
  }
```

Reply:
This is a great suggestion, will adjust it everywhere.

Reply:
Added in d34283b2d8ef06f2cddb0acb26c366d14a572c2e. One thing to note is the construction of the response is now slightly more awkward, and kind of hard to work around as the interface for `.Result` is private.